test/test.c

```diff
@@ -31470,6 +31470,140 @@ void test_KeyValueKeys(void)
     JS_TEARDOWN;
 }
 
+void test_KeyValueKeysWithFilters(void)
+{
+    natsStatus          s;
+    kvStore             *kv = NULL;
+    kvKeysList          l;
+    char                *k = NULL;
+    kvConfig            kvc;
+    kvWatchOptions      o;
+    int                 i;
+    const char **defaultSubject = (const char *[]){">"};
+    l.Count = 0;
+    l.Keys = NULL;
+
+    JS_SETUP(2, 10, 14);
+
+    test("Create KV: ");
+    kvConfig_Init(&kvc);
+    kvc.Bucket = "KVS";
+    kvc.History = 2;
+    s = js_CreateKeyValue(&kv, js, &kvc);
+    testCond(s == NATS_OK);
+
+    test("Populate: ");
+    s = kvStore_PutString(NULL, kv, "a.b", "a.b");
+    IFOK(s, kvStore_PutString(NULL, kv, "a.d", "a.d"));
+    IFOK(s, kvStore_PutString(NULL, kv, "c.d", "c.d"));
+    IFOK(s, kvStore_PutString(NULL, kv, "e.f", "e.f"));
+    IFOK(s, kvStore_PutString(NULL, kv, "e.a.f", "e.a.f"));
+    testCond(s == NATS_OK);
+
+    test("Get keys with filters (bad args): List is null");
+    // list is NULL
+    s = kvStore_KeysWithFilters(NULL, kv, NULL,defaultSubject,1);
+    testCond((s == NATS_INVALID_ARG) && (l.Keys == NULL) && (l.Count == 0));
```

Comment:
Since you don't pass `l`, those checks (keys and count) are not needed.

Reply:
Good catch, changed and tested this. Thanks for this.